src/charm.py

```diff
@@ -35,83 +79,515 @@ def __init__(self, *args: typing.Any):
-                container.replan()
-
-                logger.debug("Log level for gunicorn changed to '%s'", log_level)
-                self.unit.status = ops.ActiveStatus()
-            else:
-                # We were unable to connect to the Pebble API, so we defer this event
-                event.defer()
-                self.unit.status = ops.WaitingStatus("waiting for Pebble API")
-        else:
-            # In this case, the config option is bad, so block the charm and notify the operator.
-            self.unit.status = ops.BlockedStatus("invalid log level: '{log_level}'")
+            health_check_url: OpenCTI health check URL
 
-    @property
-    def _pebble_layer(self) -> pebble.LayerDict:
-        """Return a dictionary representing a Pebble layer."""
-        return {
-            "summary": "httpbin layer",
-            "description": "pebble config layer for httpbin",
-            "services": {
-                "httpbin": {
+        Returns:
+            The service part of OpenCTI pebble plan
+        """
+        worker_service: ops.pebble.ServiceDict = {
+            "override": "replace",
+            "command": "python3 worker.py",
+            "working-dir": "/opt/opencti-worker",
+            "environment": {
+                "OPENCTI_URL": "http://localhost:8080",
+                "OPENCTI_TOKEN": self._get_peer_secret(self._PEER_SECRET_ADMIN_TOKEN_SECRET_FIELD),
+                "WORKER_LOG_LEVEL": "info",
+            },
+            "after": ["platform"],
+            "requires": ["platform"],
+        }
+        return ops.pebble.LayerDict(
+            summary="OpenCTI platform/worker",
+            description="OpenCTI platform/worker",
+            services={
+                "charm-callback": {
+                    "override": "replace",
+                    "command": f"bash {self._install_callback_script(health_check_url)}",
+                },
+                "platform": {
                     "override": "replace",
-                    "summary": "httpbin",
-                    "command": "gunicorn -b 0.0.0.0:80 httpbin:app -k gevent",
-                    "startup": "enabled",
+                    "command": "node build/back.js",
+                    "working-dir": "/opt/opencti",
                     "environment": {
-                        "GUNICORN_CMD_ARGS": f"--log-level {self.model.config['log-level']}"
+                        "NODE_OPTIONS": "--max-old-space-size=8096",
+                        "NODE_ENV": "production",
+                        "PYTHONUNBUFFERED": "1",
+                        "APP__PORT": "8080",
+                        "APP__APP_LOGS__LOGS_LEVEL": "info",
+                        "PROVIDERS__LOCAL__STRATEGY": "LocalStrategy",
+                        "APP__TELEMETRY__METRICS__ENABLED": "true",
+                        **self._gen_secret_env(),
+                        **self._prepare_opensearch_env(),
```

Comment:
Let's try and move the side effects and integration checks somewhere else

Reply:
Create a new function named `_install_opensearch_cert` to install OpenSearch TLS certificates when necessary.